### PR TITLE
Fix service failure in vmware

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@aruniiird , @GowthamShanmugam , @shtripat Please review

**Comment by TimothyAsirJeyasing** [comment]
@aruniiird , @GowthamShanmugam , @shtripat Please review

File: tendrl/node_agent/node_sync/disk_sync.py [hunk 1]
```diff
@@ -231,6 +234,9 @@ def get_disk_details():
                     if "by-id/virtio" in entry:
                         devlist['disk_id'] = entry.split('/')[-1]
                         break
+            elif "VMware" in devlist["vendor"]:
+                 devlist["disk_id"] = (devlist["vendor"] + "_" +
+                    devlist["device"] + "_" + devlist["hardware_id"])
```
Comment (aruniiird) [inline tendrl/node_agent/node_sync/disk_sync.py:239]:
```suggestion
                 devlist["disk_id"] = "{vendor}_{device}_{hardware_id}".format(**devlist)
```
makes it much efficient than the `+` concatenation.

**Review by aruniiird — CHANGES_REQUESTED** [review]
A minor change request.

**Review by aruniiird — COMMENTED** [review]
We are generally ignoring the errors (referring to the unicode conversion part), is that fine?

File: tendrl/node_agent/node_sync/disk_sync.py [hunk 2]
```diff
@@ -231,6 +238,9 @@ def get_disk_details():
                     if "by-id/virtio" in entry:
                         devlist['disk_id'] = entry.split('/')[-1]
                         break
+            elif "VMware" in devlist["vendor"]:
+                 devlist["disk_id"] = \
+                    "{vendor}_{device}_{parent_id}{hardware_id}".format(**devlist)
```
Comment (aruniiird) [inline tendrl/node_agent/node_sync/disk_sync.py:243]:
```suggestion
                    "{vendor}_{device}_{parent_id}_{hardware_id}".format(**devlist)
```

**Review by aruniiird — APPROVED** [review]

**Pull request merged**